writeengine/server/we_dmlcommandproc.cpp

```diff
@@ -914,34 +993,38 @@ uint8_t WE_DMLCommandProc::processBatchInsert(messageqcpp::ByteStream& bs, std::
         colDBRootExtentInfo.push_back(dbRootExtent);
 
         Column aColumn;
-        aColumn.colWidth = convertor.getCorrectRowWidth(colType.colDataType, colType.colWidth);
-        aColumn.colDataType = colType.colDataType;
-        aColumn.compressionType = colType.compressionType;
-        aColumn.dataFile.oid = ridList[i].objnum;
-        aColumn.dataFile.fPartition = dbRootExtent.fPartition;
-        aColumn.dataFile.fSegment = dbRootExtent.fSegment;
-        aColumn.dataFile.fDbRoot = dbRootExtent.fDbRoot;
-        aColumn.dataFile.hwm = dbRootExtent.fLocalHwm;
-        columns.push_back(aColumn);
 
-        if ((colType.compressionType > 0) && (colType.ddn.dictOID > 0))
+        if ((i == ridList.size() - 1) && (tableAUXColOid > 3000)) // AUX column
         {
-          DctnryStruct aDctnry;
-          aDctnry.dctnryOid = colType.ddn.dictOID;
-          aDctnry.fColPartition = dbRootExtent.fPartition;
-          aDctnry.fColSegment = dbRootExtent.fSegment;
-          aDctnry.fColDbRoot = dbRootExtent.fDbRoot;
-          dctnryList.push_back(aDctnry);
-        }
-
-        if (colType.ddn.dictOID > 0)
-        {
-          dctnryStoreOids[i] = colType.ddn.dictOID;
+          aColumn.colWidth = execplan::AUX_COL_WIDTH;
+          aColumn.colDataType = execplan::AUX_COL_DATATYPE;
+          // TODO MCOL-5021 compressionType for the AUX column is hard-coded to 2
+          aColumn.compressionType = execplan::AUX_COL_COMPRESSION_TYPE;
         }
         else
         {
-          dctnryStoreOids[i] = 0;
+          const CalpontSystemCatalog::ColType& colType = colTypes[i];
+          aColumn.colWidth = convertor.getCorrectRowWidth(colType.colDataType, colType.colWidth);
+          aColumn.colDataType = colType.colDataType;
+          aColumn.compressionType = colType.compressionType;
+
+          if ((colType.compressionType > 0) && (colType.ddn.dictOID > 0))
+          {
+            DctnryStruct aDctnry;
+            aDctnry.dctnryOid = colType.ddn.dictOID;
+            aDctnry.fColPartition = dbRootExtent.fPartition;
+            aDctnry.fColSegment = dbRootExtent.fSegment;
+            aDctnry.fColDbRoot = dbRootExtent.fDbRoot;
+            dctnryList.push_back(aDctnry);
```

Comment:
I would replace this block with a single line dctnryList.emplace_back({colType.ddn.dictOID...}); 
But this is up to you to decide.

Reply:
The existing code is more readable. `WriteEngine::DctnryStruct dctnryStruct;` calls the default ctor which initialises the struct members to some default values, followed by explicitly setting a subset of those members to other values here. Vs with the `emplace_back()` approach, we will have to interleave user-supplied values with defaults such as `dctnryList.emplace_back({colType.ddn.dictOID, 0, 0, dbRootExtent.fPartition ... etc.});` which makes it harder to read.

Reply:
You can add a custom ctor for the purpose though.

Reply:
I would leave this as is, it goes outside the scope of this patch.